learning_journal_basic1/tests.py

```diff
@@ -1,29 +1,162 @@
-import unittest
+import pytest
+import transaction
+import datetime
+from ..views.default import TIME_FORMAT
 
 from pyramid import testing
 
+from .models import (
+    MyModel,
+    get_engine,
+    get_session_factory,
+    get_tm_session,
+)
+from .models.meta import Base
 
-class ViewTests(unittest.TestCase):
-    def setUp(self):
-        self.config = testing.setUp()
+DB_SETTINGS = {'sqlalchemy.url': 'sqlite:////tmp/testme.sqlite'}
 
-    def tearDown(self):
+
+@pytest.fixture(scope="session")
+def sqlengine(request):
+    """Set up sql engine for testing."""
+    config = testing.setUp(settings=DB_SETTINGS)
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
         testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    """Set up new session for testing."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    """Set up mock database for testing."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyModel(title="Purgatory", body="The reality of pythonic purgatory cannot be denied.", date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyModel).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Testing a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
```

Comment:
just an FYI.  If your `dummy_request` fixture already relies on `new_session`, and you then attach the session to the request (as you do in line 69 above), then you don't really need to depend on `new_session` directly in this test.  You can get to the session in question by using `request.dbsession`, so there's no need to have a second handle on that database session.  One is enough here.